app/models/project.rb

```diff
@@ -729,9 +729,10 @@ def check_status
   def rate_limited_npm_request(url)
     # This key needs to be host-specific so hosts don't overwrite each other's different values.
     key = "#{Project::CHECK_STATUS_NPM_RETRY_AFTER}:#{Socket.gethostname}"
-    current_retry_after = REDIS.get(key)
+    # The value will be the original 'retry-after', and the TTL will be the remaining 'retry-after', so use the latter.
+    current_retry_after = REDIS.ttl(key)
 
-    if current_retry_after.present?
+    if current_retry_after >= 0 # -2 == key does not exist, -1 == key exists without a TTL
```

Comment:
Can `current_retry_after` ever be nil or empty or some other non-integer?

Reply:
nah, the docs say it'll only return a signed int: https://redis.io/docs/latest/commands/ttl/#resp2resp3-reply

Reply:
ok, just making sure we aren't going to get a pile of errors when their stuff goes down and we end up with a bunch of errors about nils